DirectProgramming/DPC++FPGA/Tutorials/Features/ac_types/ap_float/README.md

```diff
@@ -0,0 +1,471 @@
+
+## Purpose
+
+This FPGA tutorial shows how to use the `ap_float` type with some simple examples and recommended best practices.
+
+This data-type can be used in place of native floating point types to generate area efficient and optimized designs for the FPGA. For example, operations which do not utilize all of the bits the native types or desings which do not require all of the range and precision of native types are good candidates for replacement with the `ap_float` type.
+
+This tutorial will present the following:
+1. How to include the `ap_float` type and an overview of common ap_float use cases.
+2. A Polynomial Sine Approximation example which illustrates how to trade off mathematical accuracy for lesser FPGA resource utilization.
+3. Rounding Mode and native type to ap_float type conversion examples which describe various `ap_float` rounding modes and their effect on accuracy and FPGA resource utilization.
+4. A Quadratic Equation Solver example which show cases explicit `ap_float` math functions and how they can be used to replace mathematical operators like `*, /, +` and `-` for better quality of results.
+
+### Simple Code Example
+
+An `ap_float` number can be defined as follows:
+
+```cpp
+ihc::ap_float<EW, MW> a;
+```
+Here EW specifies the exponent width and MW specifies the mantissa width of the number. Optionally, another template parameter can be specified to set the rounding mode. For more details please refer to the section titled `Variable-Precision Integer and Floating-Point Support` in the Intel® oneAPI DPC++ FPGA Optimization Guide.
+
+To use this type in your code, you must include the following header:
+
+```cpp
+#include <sycl/ext/intel/ac_types/ap_float.hpp>
+```
+
+To use `ap_float` math functions, you must include the following header:
+
+```cpp
+#include <sycl/ext/intel/ac_types/ap_float_math.hpp>
+```
+
+Additionally, you must use the flag `-qactypes` in order to ensure that the headers are correctly included and that the compiler links against the necessary libraries for emulation support.
+
+You can easily convert your existing designs that use native floating-point types to use `ap_float`: simply switch the original type. For math functions, `ap_float` has the "ihc_" prefix, you can simply switch your math functions accordingly, e.g. `sin(x)` should be changed to `ihc_sin(x)` for `ap_float`.
+
+After the migration, you can use the area report to examine the area improvement of your design. In general, the line structure of the area report does not change. For example, instead of seeing a `X bit floating-point multiply` on the old design, the source line for the changed design would show `fpga.vpfp.mul`. 
+
+You should confirm that the area used for the operation has indeed decreased from a Quartus compile. You should also make sure that the result of your design still meets your accuracy expectations through simulation.
+
+### Overview of common use cases for ap_float
```

Comment:
some of your subtitles are Fully Capitalized For All Words, and Some only capitalize the first word.

Since the subtitles in this tutorial go up to 4 levels down, I think you should go with having all words capitalized because it makes it easier to tell a subtitle from text.